### DISCUSSION TITLE
Building terminal.c from sources causes ~T~B characters to show up

### DISCUSSION THREAD

**Discussion by jgarte** [lem-project/lem#1567] [Q&A]
Hi,

I tried building `terminal.c` with `gcc` and guix instead of using the [pre-compiled shared object binary files](https://github.com/lem-project/lem/tree/main/extensions/terminal/lib) in the current repo and I get the below `~T~B` terminal characters.

I am opening this discussion to see if someone has any ideas for how I can debug what the issue might be exactly.

Below are two screenshots of the issue:

![image](https://github.com/user-attachments/assets/e08cdf87-6c26-48f0-9601-f545863eee9a)

![image](https://github.com/user-attachments/assets/e24a1fff-4d0c-4148-bbf7-9165666be378)

Here is the Guix package definition that shows the `gcc` command that I called to compile `terminal.c` along with all the dependencies that are available for the package:

https://git.savannah.gnu.org/cgit/guix.git/tree/gnu/packages/text-editors.scm?id=master#n194

**Comment by jgarte** [comment]
I will try soonish to compile `terminal.c` without Guix on Fedora to see if anything changes in that bug that I am experiencing.

**Comment by cxxxr** [comment]
https://github.com/lem-project/lem/blob/7b380ea04cea2696975da6c7d8c29e57ce2907a2/frontends/ncurses/style.lisp#L47
I think your terminal can't see these characters.

What happens if you change the LANG environment variable?
```
TERM=xterm-256color
LANG=en_US.UTF-8
```

**Reply by jgarte** [reply]
![image](https://github.com/user-attachments/assets/f4417916-025a-44bf-ba91-9f4a18241e78)

**Comment by jgarte** [comment]
The odd thing is that if I create an empty [guix shell](https://guix.gnu.org/manual/en/html_node/Invoking-guix-shell.html), and run lem inside it, the characters go away...

**Reply by jgarte** [reply]
I tried diffing the difference between my regular shell and the `guix shell` subshell but the only thing I can tell that is different is that `SHLVL` is incremented by 1 🫠 

`SHLVL=1` -> `SHLVL=2`

**Reply by jgarte** [reply]
I also get the same `~T~B` characters with `lem --interactive sdl2`:

![image](https://github.com/user-attachments/assets/00b238a3-5d34-4cee-8e5f-b0aa8512c414)
